### PR TITLE
refactor(http): Extract write handler logic for re-use elsewhere

### PR DESCRIPTION
**Pull request by brettbuddin**
This is the first PR in a 3-part story to deliver an API for 1.x writes.

This shuffles the structure of the existing `/api/v2/write` endpoint to allow the `/write` (1.x) endpoint to share the point parse and writing code. Code that was originally placed directly in the main HTTP handler has been moved out into other supporting structures for re-use.

📎 influxdata/idpe#6948

### CODE REVIEW

File: http/api_handler.go [hunk 1]
```diff
@@ -208,9 +209,11 @@ func NewAPIHandler(b *APIBackend, opts ...APIHandlerOptFn) *APIHandler {
...
 	h.Mount(prefixWrite, NewWriteHandler(b.Logger, writeBackend,
 		WithMaxBatchSizeBytes(b.MaxBatchSizeBytes),
-		WithParserMaxBytes(b.WriteParserMaxBytes),
-		WithParserMaxLines(b.WriteParserMaxLines),
-		WithParserMaxValues(b.WriteParserMaxValues),
+		WithParserOptions(
+			models.WithParserMaxBytes(b.WriteParserMaxBytes),
+			models.WithParserMaxLines(b.WriteParserMaxLines),
+			models.WithParserMaxValues(b.WriteParserMaxValues),
+		),
```
Comment (stuartcarnie) [inline http/api_handler.go:216]: 👏

File: kit/io/limited_read_closer.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+			err = cerr
+		}
+
+		// only call close once
+		l.close = func() error { return nil }
+	}()
+
+	if l.N < 1 {
+		l.err = ErrReadLimitExceeded
+	}
```
Comment (stuartcarnie) [inline kit/io/limited_read_closer.go:45]: What happens if the source data was exactly `N` bytes? Would this report an error even though it had successfully read all data?

Comment (brettbuddin) [reply kit/io/limited_read_closer.go:45]: Oooo great eye. I think we should make that less than 0 instead.

Comment (brettbuddin) [reply kit/io/limited_read_closer.go:45]: Nevermind. It looks like they did this intentionally. In the constructor they initialize the `io.LimitReader` with `n + 1`. That's consistent with the `l.N < 1` here. I'm going to leave it alone, but write a few tests to clamp in the behavior.

**Pull request merged**